src/setupTests.ts

```diff
@@ -3,3 +3,42 @@
 // expect(element).toHaveTextContent(/react/i)
 // learn more: https://github.com/testing-library/jest-dom
 import "@testing-library/jest-dom";
+
+import * as fs from "fs";
+import * as path from "path";
+import * as jsii from "@jsii/spec";
+import * as reflect from "jsii-reflect";
+
+// randomly selected...
+const ASSEMBLY_UNDER_TEST = "@aws-cdk/aws-ecr";
+
+function createAssembly(): reflect.Assembly {
+  const ts = new reflect.TypeSystem();
+
+  const packages = `${__dirname}/__assemblies__`;
+
+  collectAssebmlies(packages, ts);
+
+  return ts.findAssembly(ASSEMBLY_UNDER_TEST);
+}
+
+// expose the type system under test gloablly
+(global as any).assembly = createAssembly();
+
+function collectAssebmlies(p: string, ts: reflect.TypeSystem) {
+  const stat = fs.statSync(p);
+
+  if (stat.isDirectory()) {
+    fs.readdirSync(p).map((f) => collectAssebmlies(path.join(p, f), ts));
+  } else {
+    if (!p.endsWith("jsii.json")) {
+      throw new Error(
+        `Unexpected file in packages: '${p}' (should contain only jsii.json files)`
+      );
+    }
+    const assembly = JSON.parse(
+      fs.readFileSync(p, { encoding: "utf8" })
```

Comment:
We should  have validation method here. To make sure the JSII file has the correct structure, to sanitize the text for security reasons etc.

Reply:
I don't think we need schema validation because these are official schema's produced by JSII, and validation takes a lot of time. 

As far as sanitation and security, yes, but this is out of scope for now I believe..?

Reply:
You are correct! It's something we will do on the backend